src/services/user.js

```diff
@@ -1,82 +1,56 @@
-const USERS = [];
-let ID = 0;
-
-function findUser(id) {
-  return USERS.find((user) => user.id === id && !user.isDeleted);
-}
-
-function formUser({ id, login, password, age }) {
-  return {
-    id,
-    login,
-    password,
-    age,
-  };
-}
-
-function getUser(id) {
-  const user = findUser(id);
-
-  if (user) {
-    return formUser(user);
+import {
+  createUser,
+  updateUser,
+  deleteUser,
+  findUserById,
+  findUsersByLogin,
+} from '../data-access/user.js';
+import GeneralError from '../errors/general.js';
+
+export default class UserService {
+  static formUser({ id, login, password, age }) {
+    return {
+      id,
+      login,
+      password,
+      age,
+    };
   }
 
-  return null;
-}
+  static async addUser({ login, password, age }) {
+    const user = await createUser(login, password, age);
 
-function addUser({ login = '', password = '', age = '' }) {
-  ID += 1;
-  USERS.push({
-    id: ID.toString(),
-    login,
-    password,
-    age,
-    isDeleted: false,
-  });
-}
+    return user ? UserService.formUser(user) : null;
+  }
 
-function deleteUser(id) {
-  const user = findUser(id);
+  static async updateUser(id, { login, password, age }) {
+    const isUserExist = !!(await findUserById(id));
 
-  if (user) {
-    user.isDeleted = true;
-    return true;
+    if (isUserExist) {
+      await updateUser(id, login, password, age);
+    } else {
+      throw new GeneralError(`User with id = ${id} cannot be updated`, 400);
```

Comment:
It is better to create more specific error. It may be useful when you want to use different error handler middlewares for different error types. 
For example: if it is UserNotFoundError -> return an error, if DataAccessError => redirect to a support page. 

Also it will be useful while implementing monitoring, so you will see what errors occurred mostly and also create alerts for some critical errors

Reply:
Created new error type `NotFoundError`.  Maybe we should change message to the previous one `User with id = ${id}' not found` as for now this the reason why we can't udate/delete user.